### DISCUSSION TITLE
[2023-09-18] GUI Team Update

### DISCUSSION THREAD

**Discussion by wdanilo** [enso-org/enso#7829] [Weekly Demo Day]
Website progress, including:
- Talking a lot with James, Ned, Sylwia about texts.
- Adding additional sections.
- Working again on spacing and metrics to be sure that everything displays correctly. We had some problems with that, they are solved, but they took some time to do it correctly.

https://github.com/enso-org/enso/assets/1623053/e7d822fa-2655-4d0e-b79b-43622000af02

**Comment by MichaelMauderer** [comment]
WIP of GUI2 text editor (integration of Monaco): 

![Peek 2023-09-18 15-12](https://github.com/enso-org/enso/assets/1428930/7175c2fd-2052-46a1-b57f-bfb406a9435e)

**Comment by vitvakatu** [comment]
Last week, I was fixing issues in the IDE.
- https://github.com/enso-org/enso/pull/7782
- https://github.com/enso-org/enso/pull/7798
- https://github.com/enso-org/enso/pull/7810
- https://github.com/enso-org/enso/pull/7779 is still not merged because of the issues spotted on QA. It turns out a lot of things on develop work by accident.

Still, a few more tasks are left, I will handle them in the following days:
- https://github.com/enso-org/enso/issues/7474
- https://github.com/enso-org/enso/issues/7824

**Comment by Frizi** [comment]
- Added reading of true language-server project data and showing it on the graph. Currently this is read-only, full synchronization still in progress. #7813
- Working on converting the graph to use true Enso AST, which is now necessary to avoid writing garbage metadata once we add read-write synchronization with language server. #7791

<img width="544" alt="image" src="https://github.com/enso-org/enso/assets/919491/c2b67bbc-d4c0-4b4b-8911-1499c2b3e73f">

**Comment by somebody1234** [comment]
I was working on:

- Implementing the binary protocol in order to connect visualizations to actual data: #7815
  - **On hold.**
  - **Not working yet.** I *think* it's pretty close, but (IIRC) for some reason `executionContext` calls were failing
- Visualizations: #7773
  - Finish porting the rest of the visualizations (missed one, which was implemented today)
  - Addressing issue(s) raised in previous demo
  - Addressing review (almost done, there is one issue with how d3 and Vue should be interacting)

<details>
  <summary>Notes on video</summary>
  So it turns out the video revealed some bugs:
  <ul>
   <li>column autosizing is not working the first time the table viz is opened</li>
   <li>the histogram bars also don't show up before resizing (understandable as it is the most recently ported viz.)</li>
   <li>the histogram's x-axis seems to not be wide enough</li>
  </ul>
</details>

https://github.com/enso-org/enso/assets/4046547/a2c1a1ad-6614-40c4-b197-00dc90b85797

**Comment by mwu-tow** [comment]
Adding support for the new GUI in the build scripts and CI, including:
* CI-time checks;
* CLI commands support;
* Electron packaging.

![obraz](https://github.com/enso-org/enso/assets/1548407/5335ca07-5250-4559-aa9c-1c01476aca82)
The PR https://github.com/enso-org/enso/pull/7832 is still draft but should be ready soon.

**Comment by kazcw** [comment]
Parser API: Implemented "serialization" to new format that will enable lazy deserialization; working on generating TS/Java deserializers that implement new API design.

**Comment by JaroslavTulach** [comment]
Engine & language server GUI related fixes:
- #7801
- #7807
- #7825 
- #7659
- #7812
- #7831

**Comment by farmaazon** [comment]
### New GUI
Behold the filtering in the Component Browser:

https://github.com/enso-org/enso/assets/3919101/3d1d0ec0-87ad-4205-ab33-6bc6f9036203

Compared to the state from previous week:
* the input is nicely styled,
* focus management improved
* The already written qualified name is taken into account 
* "Show unstable" button is implemented
 
It's almost merged, only a last tiny review comment is to be applied. Then I will implement loading and updating suggestion database from language server (in the screencast, I just load a static snapshot from the old GUI).

### Old GUI

Checked the engine fix for operators' parameter names. Engine indeed work, but IDE still does not display placeholders for nodes like `2 + `. Found the cause and added a test, the only thing left is to actually fix it.
